cpp/src/arrow/array.h

```diff
@@ -529,13 +529,14 @@ class ARROW_EXPORT ListArray : public Array {
   }
 
  protected:
-  // this constructor defers SetData to a derived array class
+  // This constructor defers SetData to a derived array class
   ListArray() = default;
   void SetData(const std::shared_ptr<ArrayData>& data);
-  const int32_t* raw_value_offsets_;
+
+  const int32_t* raw_value_offsets_ = nullptr;
 
  private:
-  const ListType* list_type_;
+  const ListType* list_type_ = nullptr;
```

Comment:
Can't use `nullptr` in headers, use the macro `NULLPTR` instead.

Reply:
:-/

Reply:
I wonder if we should keep this... https://en.wikipedia.org/wiki/C%2B%2B/CLI

Reply:
I think that C++/CLI is the new defacto way for interfacing with C++ from C#.NET. I wrote a linter to check for nullptr and other issues, so I don't think it's a huge issue